GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -726,21 +737,18 @@ public int Compare(string a, string b)
 
                 int IndexOf(string s)
                 {
-                    var head = s.StartsWith("remotes/") ? "refs/" : _prefix;
-                    var tail = s;
-                    var headLength = head.Length;
-                    var length = headLength + s.Length;
-
-                    for (var i = 0; i < _otherList.Count; i++)
+                    if (s.StartsWith("remotes/"))
```

Comment:
Can the user of this comparer persist the data in sanitised form so we don't have to allocate temporary strings here?

Reply:
I don't know. Could you try to convince me that it is worth to care about this? These are short-lived strings and AFAIK the GC deals pretty well with them. The gain in performance caused by avoiding string allocations is almost none if any at all (The old version of this method with string allocations seems to be faster - didn't do a benchmark, just compared freezes times when navigating through blame lines). The gain achieved by replacing List with Dictionary is significant, noticeable with a bare eye. There are many other places where optimization could be done and I would rather optimize them before doing so low level optimizations like avoiding allocations.

Reply:
@jbialobr I don't doubt there's already a gain, the implementation looks great. I'm just wary of comparers that allocate. I don't even know how this comparer is used, but if it was impacting perf due to algorithmic complexity, then presumably there were quite a few calls. Right now it's allocating temp strings for each comparison. If the inputs were all canonical (`ref/` form or whatever) then you could eliminate the allocations and the `StartsWith` call.